mobile/components/Login/style.js

```diff
@@ -0,0 +1,172 @@
+// @flow
+import React from 'react';
+import styled from 'styled-components/native';
+import { View, TouchableOpacity, ScrollView } from 'react-native';
+import { Svg } from 'expo';
+const { Path, G } = Svg;
+import theme from '../theme';
+import Anchor from '../Anchor';
+
+export const Container = (props: any) => {
+  const Container = styled.View`
+    padding: 32px 16px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  `;
+
+  return (
+    <ScrollView>
+      <Container>{props.children}</Container>
+    </ScrollView>
+  );
+};
+
+export const Emoji = () => (
+  <Svg width="64" height="64" viewBox="0 0 32 32">
+    <G>
+      <Path
+        fill={theme.text.alt}
+        fillRule="evenodd"
+        d="M26,16c0,5.523 -4.477,10 -10,10c-5.523,0 -10,-4.477 -10,-10c0,-5.523 4.477,-10 10,-10c5.523,0 10,4.477 10,10Zm2,0c0,6.627 -5.373,12 -12,12c-6.627,0 -12,-5.373 -12,-12c0,-6.627 5.373,-12 12,-12c6.627,0 12,5.373 12,12Zm-17.5,0c0.829,0 1.5,-0.672 1.5,-1.5c0,-0.828 -0.671,-1.5 -1.5,-1.5c-0.829,0 -1.5,0.672 -1.5,1.5c0,0.828 0.671,1.5 1.5,1.5Zm12.5,-1.5c0,0.828 -0.671,1.5 -1.5,1.5c-0.829,0 -1.5,-0.672 -1.5,-1.5c0,-0.828 0.671,-1.5 1.5,-1.5c0.829,0 1.5,0.672 1.5,1.5Zm-6.999,8c2.45,0 4.534,-1.715 5,-4c0.232,-1.14 -3,-1.5 -5,-1.5c-2,0 -5.259,0.231 -5,1.5c0.466,2.285 2.549,4 5,4Zm2,-2c0,0 -0.896,0.5 -2,0.5c-1.105,0 -2,-0.5 -2,-0.5c0,0 0.895,-1 2,-1c1.104,0 2,1 2,1Z"
+      />
+    </G>
+  </Svg>
+);
+
+export const Title = styled.Text`
+  color: ${props => props.theme.text.default};
+  font-size: 40px;
+  font-weight: 900;
+  text-align: center;
+`;
+
+export const Subtitle = styled.Text`
+  color: ${props => props.theme.text.alt};
+  font-weight: 500;
+  font-size: 16px;
+  margin-top: 16px;
+  margin-bottom: 16px;
+  text-align: center;
+  padding: 0px 32px;
+`;
+
+type SigninButtonProps = {
+  color: string,
+  children: React$Node,
+  onPress: Function,
+};
+const SigninButton = ({ color, children, onPress }: SigninButtonProps) => {
+  const Container = styled.TouchableOpacity`
+    width: 220px;
+    height: 51px;
+    display: flex;
+    flex-direction: row;
+    justify-content: center;
+    align-items: center;
+    border-radius: 8px;
+    padding: 8px;
+    margin-top: 16px;
+    padding-right: 16px;
+    background-color: ${color};
```

Comment:
Please move this out of the `SigninButton`—the way this is built right now, the `Container` component will be recreated on every render. That's very expensive!

This is what it should look like:

```JS
const Container = styled.TouchableOpacity`
  background-color: ${props => props.color};
  // ...
`

const SigninButton = ({ color, children, onPress }: SigninButtonProps) => {
  return <Container onPress={onPress} color={color}>{children}</Container>;
}
```

Reply:
@mxstbr I just fix it!
I'v not noticed this performance issue until now.
thank you for your insightful review:smile: